packages/data-service/src/data-service.ts

```diff
@@ -2280,6 +2281,7 @@ class DataServiceImpl extends WithLogContext implements DataService {
         this._logger.info(mongoLogId(1_001_000_020), 'Server closed', {
           address: evt.address,
         });
+        this._emitter.emit('serverClosed');
```

Comment:
How will we be using this event in VSCode?

For https://github.com/mongodb-js/vscode/pull/734, we'll need the `client.on('close')` event, not `serverClosed` (which can be _any_ server of the topology, e.g. a replica set member being taken down for maintenance while the replica set as a whole remains available)

Reply:
I think we can do `this._activeDataService?.once('serverClosed', ...)`?